recipes/qt/5.x.x/conanfile.py

```diff
@@ -695,6 +696,20 @@ def package_info(self):
                 self.cpp_info.frameworks.extend(["Cocoa"])    # "libQt5Core.a" require "_OBJC_CLASS_$_NSApplication" and more, which are in "Cocoa" framework
                 self.cpp_info.frameworks.extend(["Security"]) # "libQt5Core.a" require "_SecRequirementCreateWithString" and more, which are in "Security" framework
 
+        tools.save(os.path.join("lib", "cmake", "Qt5Core", "extras.cmake"),
+                    "set(Qt5Core_QMAKE_EXECUTABLE ${CMAKE_CURRENT_LIST_DIR}/../../../bin/qmake)\n"
+                    "set(Qt5Core_MOC_EXECUTABLE ${CMAKE_CURRENT_LIST_DIR}/../../../bin/moc)\n"
+                    "set(Qt5Core_RCC_EXECUTABLE ${CMAKE_CURRENT_LIST_DIR}/../../../bin/rcc)\n"
+                    "set(Qt5Core_UIC_EXECUTABLE ${CMAKE_CURRENT_LIST_DIR}/../../../bin/uic)")
```

Comment:
Request for comments: With the change as it is now, the user has to manually call `qt5_wrap_cpp` on the sources in the `CMakeLists.txt`. Most projects do not use this, but instead rely on `CMAKE_AUTOMOC`, which automatically invokes MOC on sources of all targets. This currently does not work, because cmake tries to invoke the target `Qt5::moc`, which does not exist. Would it be fine if  I added this `set_target_properties(Qt5::moc PROPERTIES IMPORTED_LOCATION ${Qt5Core_QMAKE_EXECUTABLE})` ? I'm not sure because all the targets generated by conan are in the `qt` namespace, not in the `Qt5` namespace, and I don't want to make this shift without components support. WDYT ?

Reply:
I think this is what is making my PR fail
https://github.com/conan-io/conan-center-index/pull/4562

Reply:
yes: take a look at https://github.com/ericriff/conan-center-index/pull/5